windows-rdp/main.test.ts

```diff
@@ -0,0 +1,130 @@
+import { describe, expect, it } from "bun:test";
+import {
+  TerraformState,
+  runTerraformApply,
+  runTerraformInit,
+  testRequiredVariables,
+} from "../test";
+
+type TestVariables = Readonly<{
+  agent_id: string;
+  resource_id: string;
+  admin_username?: string;
+  admin_password?: string;
+}>;
+
+function findWindowsRpdScript(state: TerraformState): string | null {
+  for (const resource of state.resources) {
+    const isRdpScriptResource =
+      resource.type === "coder_script" && resource.name === "windows-rdp";
+
+    if (!isRdpScriptResource) {
+      continue;
+    }
+
+    for (const instance of resource.instances) {
+      if (instance.attributes.display_name === "windows-rdp") {
+        return instance.attributes.script;
+      }
+    }
+  }
+
+  return null;
+}
+
+/**
+ * @todo It would be nice if we had a way to verify that the Devolutions root
+ * HTML file is modified to include the import for the patched Coder script,
+ * but the current test setup doesn't really make that viable
+ */
+describe("Web RDP", async () => {
+  await runTerraformInit(import.meta.dir);
+  testRequiredVariables<TestVariables>(import.meta.dir, {
+    agent_id: "foo",
+    resource_id: "bar",
+  });
+
+  it("Has the PowerShell script install Devolutions Gateway", async () => {
+    const state = await runTerraformApply<TestVariables>(import.meta.dir, {
+      agent_id: "foo",
+      resource_id: "bar",
+    });
+
+    const lines = findWindowsRpdScript(state)
```

Comment:
What if this function returns `null`?

Reply:
Oh, good catch. The modules codebase doesn't have a super strict TypeScript compiler, so some of the types aren't as accurate as they should be, and some null/undefined values aren't being surfaced when they should be

I was already planning to tighten things up, but I for sure need to handle the null case here